llvm/lib/Target/AMDGPU/SILoadStoreOptimizer.cpp

```diff
@@ -2571,3 +2580,19 @@ bool SILoadStoreOptimizer::runOnMachineFunction(MachineFunction &MF) {
 
   return Modified;
 }
+
+PreservedAnalyses
+SILoadStoreOptimizerPass::run(MachineFunction &MF,
+                              MachineFunctionAnalysisManager &MFAM) {
+  MFPropsModifier _(*this, MF);
```

Comment:
Why is this not used in all passes that have `getRequiredProperties` and is this required?

Reply:
Not all passes in CodeGen pipeline set them although some of them should. `MFPropsModifier ` is new pass manager only, we should ensure `PassName(...).run(...)` also set properties correctly. If pass require some properties, just use it or do something manually.

Reply:
If `MFPropsModifier` is new pass only, it seems that legacy passes do not use this either.

Should I add this to the legacy pass as well? Which is:

```cpp
bool GCNDPPCombineLegacy::runOnMachineFunction(MachineFunction &MF) {
  if (skipFunction(MF.getFunction()))
    return false;

  MFPropsModifier _(*this, MF);
  return GCNDPPCombine().run(MF);
}
```

Reply:
There is no need to use them in legacy pass manager version. In legacy pass manager, `getRequiredProperties` etc. are virtual method and `MachineFunctionProperties` are handled by `MachineFunctionPass`(see `MachineFunctionPass::doInitialization` and `MachineFunctionPass::runOnFunction`).

Reply:
Got it, thanks